feat(Helm Chart): Make service.externalTrafficPolicy avaliable

at-blacknight opened the pull request:
<!--

CONTRIBUTORS GUIDE: https://github.com/grafana/alloy/blob/main/docs/developer/contributing.md#updating-the-changelog

If this is your first PR or you have not contributed in a while, we recommend
taking the time to review the guide. It gives helpful instructions for
contributors around things like how to update the changelog.

-->

#### PR Description
Minor changes in order to be able to set alloy service externalTrafficPolicy through the helm chart.

#### Which issue(s) this PR fixes

<!-- Uncomment the following line if you want that GitHub issue gets automatically closed after merging the PR -->
Fixes #4510

#### Notes to the Reviewer
Very green to OSS PR's in general so please be kind, let me know what I should do differently.

#### PR Checklist

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ x] CHANGELOG.md updated
- [ x] Documentation added

at-blacknight committed:
fix: Update values.yaml to include default externalTrafficPolicy value

at-blacknight committed:
fix: Update service.yaml to include externalTrafficPolicy

at-blacknight committed:
fix: Update README.md to include service.externalTrafficPolicy

at-blacknight committed:
Update CHANGELOG.md for #4510

at-blacknight committed:
Update CHANGELOG.md correct allocation of change catergory

kalleep reviewed (dismissed):
LGTM

kalleep commented:
Pretty sure you need to re-generate helm test files `make generate-helm-test` and possibly docs `make generate-helm-docs`

at-blacknight committed:
chore(helm): regenerate helm test fixtures for externalTrafficPolicy

at-blacknight committed:
chore(helm): regenerate chart README for externalTrafficPolicy

at-blacknight committed:
fix(helm): only emit service.externalTrafficPolicy for LoadBalancer/NodePort

Kubernetes rejects Service manifests with externalTrafficPolicy set on
ClusterIP services with: 'may only be set for externally-accessible
services'. Wrap the field in a conditional so it is only emitted when
service.type is LoadBalancer or NodePort.

Adds a ci test case (service.type=NodePort + externalTrafficPolicy=Local)
to exercise the new conditional.

at-blacknight committed:
docs(helm): collapse duplicate Enhancements heading in CHANGELOG

at-blacknight commented:
@kalleep Thanks for getting the CI going. 

Seems to be passing checks now. 

I'm unsure if we should be silently swallowing the external traffic policy when configured for clusterIP (how I currently have it)
OR
error when attempting to configure externalTrafficPolicy when using clusterIP

not sure which way to lean.

at-blacknight committed:
Merge branch 'main' into main

kalleep committed:
update tests
